JJDZR6JTH-00

PDELIMATA opened the pull request.

agatamieniuk committed:
Add getAllRecipes and FindByName in RecipeController.class

agatamieniuk committed:
Add FindByTime in RecipeController.class

agatamieniuk committed:
findingRecipeByName() in RecipeController.class fixed - search successfully with a partial word

agatamieniuk committed:
Correct method in RecipeController.class

agatamieniuk committed:
Work in progress commit

agatamieniuk committed:
Create Entities from Recipe.class

agatamieniuk committed:
Add lazy load in application.properities

agatamieniuk committed:
Fixed recipe.preparationTime in recipes.html

agatamieniuk committed:
Add Składniki column in recipes.html

agatamieniuk committed:
Adding new recipe to database method in RecipeController.class

agatamieniuk committed:
Add edit_recipe.html

agatamieniuk committed:
Add getRecipeById() and deleteRecipebyId() in RecipeService.class

agatamieniuk committed:
Add RecipeController method

agatamieniuk committed:
Add Product @Entity, @Service @Controller and @Repository

agatamieniuk committed:
Add List<Product> to Recipe.class (getter/setter/addProduct method)

agatamieniuk committed:
Add Add/RemoveProduct in RecipeController

agatamieniuk committed:
Adding only one products(thymeleaf) in recipe.html

agatamieniuk committed:
Change in List - Recipe

agatamieniuk committed:
Change List<Recipe>

agatamieniuk committed:
Work in progress - recipe/recipe_form

PDELIMATA committed:
try to fix adding product list to recipes - not finished

PDELIMATA committed:
RevipeService method save fixed.

agatamieniuk committed:
Add @RequestMapping in Recipes Controller

agatamieniuk committed:
Change create_recipe.html - add remove button and visual changes

agatamieniuk committed:
Update recipes - loading an existing recipe

agatamieniuk committed:
Add TODO to RecipeController

agatamieniuk committed:
Add findRecipeBy in RecipeRepository and getSearchRecipe in RecipeService

agatamieniuk committed:
Add searchRecipeList controller

agatamieniuk committed:
Change HTML in search-recipe.html

agatamieniuk committed:
Change application.properities

agatamieniuk committed:
Add OR query in RecipeRepository

agatamieniuk committed:
Add searching bar in search-recipe.html

agatamieniuk committed:
Add searching bar in search-recipe.html

agatamieniuk committed:
Change query in RecipeRepository to CONCAT

agatamieniuk committed:
Change Recipes navbar in HTML

agatamieniuk committed:
Add clearFilter() function in search-recipe.html

agatamieniuk committed:
Change application.properties

agatamieniuk committed:
Change in search-recipe.html

agatamieniuk committed:
Remove products from Recipe

agatamieniuk committed:
Add Left join fetch to RecipeRepository

agatamieniuk committed:
Add two times and add null in List<Product> in Recipe

agatamieniuk committed:
Work in progress -> update list and add products

agatamieniuk committed:
Add two times with null.

agatamieniuk committed:
Change findRecipeBy() method - RecipeRepository

agatamieniuk committed:
Add spring-boot-starter-validation to pom.xml

agatamieniuk committed:
Add validate to name, preparation time and description field in Recipe.class

agatamieniuk committed:
Change navbar in create-recipe.html

agatamieniuk committed:
Change binding result in create-recipe.html

agatamieniuk committed:
Add required to product in create/edit-recipe.html

agatamieniuk committed:
Add footer and prev in resources.

agatamieniuk committed:
Half of food preferences and recipe

agatamieniuk committed:
Change HTML view - foodpreferences

agatamieniuk committed:
Add translation in foodpreferences.html

agatamieniuk committed:
Comment setProductList and change html (delete required in fields)

PDELIMATA committed:
It's working so I think not so bad ;)

PDELIMATA committed:
Minor fix to app.

PDELIMATA committed:
Resolved conflicts.

szablewski commented on src/main/java/com/infoshareacademy/App.java:
```diff
@@ -1,31 +1,13 @@
+//        Menu menu = new Menu();
+//        menu.mainMenu();
+//    }
+//}
```
Jaką to pełni funcje w aplikacji ? Czy ta klasa jest w ogóle potrzebna ?

szablewski commented on pom.xml:
```diff
@@ -21,13 +21,62 @@
+			<groupId>com.vladmihalcea</groupId>
+			<artifactId>hibernate-types-5</artifactId>
+			<version>2.16.2</version>
+		</dependency>
```
Zbędna dependencja, generuje tylko zbędne logi.

szablewski commented on src/main/java/com/infoshareacademy/JavaTheHuttApplication.java:
```diff
@@ -1,13 +1,23 @@
+    @Override
+    public void run(String... args) {
 
+    }
```
Do czego służy ta linia? Co ona robi?

szablewski commented on src/main/java/com/infoshareacademy/JavaTheHuttApplication.java:
```diff
@@ -1,13 +1,23 @@
-public class JavaTheHuttApplication {
+public class JavaTheHuttApplication implements CommandLineRunner {
+
+    public static void main(String[] args) throws IOException {
```
Niepotrzebne rzucanie wyjątku.

Commit:
[61][BugFix] fix update product.

- Fixed updat product for a single object.
- Moved lists recipe products to dedicated template.
- Moved product method from RecipeController to ProductController.
- Change cascadeType in ProductRecipe.
- Added ProductInFridgeRepository.
- Generally refactoring.

Commit:
Merge pull request #16 from infoshareacademy/JJDZR6JTH-61

[61][BugFix] fix update product.

szablewski commented on src/main/java/com/infoshareacademy/controller/FoodPreferencesController.java:
```diff
@@ -1,4 +1,68 @@
+        return "recipebyfoodpreferences";
+    }
+
+    @GetMapping("/foodpreferences/set")
```
w adresie ulr nie stosujemy nazewnictwa wskazujacego na rodzaj funkji CRUD, powinien mieć formę rzeczownika

szablewski commented on src/main/java/com/infoshareacademy/controller/FoodPreferencesController.java:
```diff
@@ -1,4 +1,68 @@
+
+    @PostMapping("/foodpreferences")
+    public String saveAllergens(@ModelAttribute("foodpreferences") FoodPreferences foodPreferences) {
+        foodPreferences.setId(1L);
```
nie ustawiamy ID na sztywno

szablewski commented on src/main/java/com/infoshareacademy/controller/FoodPreferencesController.java:
```diff
@@ -1,4 +1,68 @@
+    @PostMapping("/foodpreferences")
+    public String saveAllergens(@ModelAttribute("foodpreferences") FoodPreferences foodPreferences) {
+        foodPreferences.setId(1L);
+        foodPreferencesService.setFoodPreferences(foodPreferences);
```
logika do serwisu

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+import javax.validation.Valid;
+
+@Controller
+@RequestMapping("fridge")
```
```suggestion
@RequestMapping("/fridge")
```

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
 public class FridgeController {
+
+    private FridgeService fridgeService;
+    private FridgeRepository fridgeRepository;
```
```suggestion
    private final FridgeService fridgeService;
    private final FridgeRepository fridgeRepository;
```

szablewski commented on src/main/java/com/infoshareacademy/controller/FoodPreferencesController.java:
```diff
@@ -1,4 +1,68 @@
-}
+
+    @Autowired
+    private FoodPreferencesService foodPreferencesService;
```
```suggestion
    private final FoodPreferencesService foodPreferencesService;
```

zmienne które wstrzykujecie najlepiej jak są finalne

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+        } else {
+            fridge = new Fridge();
+            fridge.setFridgeId(fridgeService.getUserId());
+        }
```
logika do serwius

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+        return "fridge";
+    }
+
+    @GetMapping("/new")
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+        return "addproductstofridge";
+    }
+
+    @PostMapping("/new")
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+        return "redirect:/fridge";
+    }
+
+    @PostMapping(value = "/new", params = {"addProduct"})
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+        return "addproductstofridge";
+    }
+
+    @PostMapping(value = "/new", params = {"removeProduct"})
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/FridgeController.java:
```diff
@@ -1,4 +1,74 @@
+        if (bindingResult.hasErrors()) {
+            return "addproductstofridge";
+        }
+        fridge.setFridgeId(1L);
```
Id na sztywno ?

szablewski commented on src/main/java/com/infoshareacademy/controller/ProductController.java:
```diff
@@ -1,4 +1,55 @@
+@Controller
 public class ProductController {
-}
+    private ProductService productService;
```
```suggestion
    private final ProductService productService;
```

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -1,4 +1,103 @@
+        return "recipes";
+    }
+
+    @GetMapping("/search")
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -1,4 +1,103 @@
+        return "search-recipe";
+    }
+
+    @GetMapping("/new")
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -1,4 +1,103 @@
+        return "create-recipe";
+    }
+
+    @PostMapping("/new")
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -1,4 +1,103 @@
+        return "redirect:/recipes";
+    }
+
+    @PostMapping(value = "/new", params = { "addProduct" })
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -1,4 +1,103 @@
+        return "create-recipe";
+    }
+
+    @PostMapping(value = "/new", params = { "removeProduct" })
```
https://nordicapis.com/10-best-practices-for-naming-api-endpoints/

szablewski commented on src/main/java/com/infoshareacademy/controller/RecipeController.java:
```diff
@@ -1,4 +1,103 @@
+        existingRecipe.setName(recipe.getName());
+        existingRecipe.setDescription(recipe.getDescription());
+
+        recipeService.saveRecipe(existingRecipe);
```
logika do serwisu

Commit:
[61][BugFix] fix update product.

- Added RequestMaping annotation with "/recipes" path

Commit:
Merge pull request #17 from infoshareacademy/JJDZR6JTH-61

[61][BugFix] fix update product.

szablewski commented on src/main/java/com/infoshareacademy/controller/ShoppingListController.java:
```diff
@@ -1,4 +1,24 @@
+    }
+
+    @GetMapping("/shoppinglist")
+    public String shoppingList(Model model) {
```
przemyslalbym jeszcze raz nazwe encji "shoppingList" oraz endpoint "/shoppinglist"

Commit:
[61][BugFix] fix update product.

- fixed delete path in recipeController

Commit:
Merge pull request #18 from infoshareacademy/JJDZR6JTH-61

[61][BugFix] fix update product.

szablewski commented on src/main/java/com/infoshareacademy/entity/recipe/Recipe.java:
```diff
@@ -1,27 +1,77 @@
+        for (ProductRecipe product : productList) {
+            if (product.getProductName().contains(s)) {
+                flag = true;
+                break;
```
break?

szablewski commented on src/main/java/com/infoshareacademy/other/Menu.java:
```diff
@@ -1,133 +1,132 @@
-package com.infoshareacademy.other;
```
jaką pełni funkcję ta klasa?

szablewski commented on src/main/java/com/infoshareacademy/repository/FoodPreferencesRepository.java:
```diff
@@ -1,4 +1,7 @@
-}
+import com.infoshareacademy.entity.food_preferences.FoodPreferences;
+import org.springframework.data.jpa.repository.JpaRepository;
+
```
Brakuje adnotacji

szablewski commented on src/main/java/com/infoshareacademy/repository/FridgeRepository.java:
```diff
@@ -1,4 +1,7 @@
+import com.infoshareacademy.entity.fridge.Fridge;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface FridgeRepository extends JpaRepository<Fridge, Long> {
```
brakuje adnotacji

szablewski commented on src/main/java/com/infoshareacademy/repository/RecipeRepository.java:
```diff
@@ -1,4 +1,16 @@
+
+import java.util.List;
+
+public interface RecipeRepository extends JpaRepository<Recipe, Long> {
```
brakuje adnotacji

szablewski commented on src/main/java/com/infoshareacademy/repository/ShoppingListRepository.java:
```diff
@@ -1,4 +1,7 @@
+import com.infoshareacademy.entity.shopping_list.ShoppingList;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface ShoppingListRepository extends JpaRepository<ShoppingList, Long> {
```
brakuje adnotacji

szablewski commented on src/main/java/com/infoshareacademy/service/FoodPreferencesService.java:
```diff
@@ -1,171 +1,100 @@
 
-    public Meat setMeatPreferences() {
-        Meat meat = new Meat();
+    private FoodPreferencesRepository foodPreferencesRepository;
+    @Autowired
+    RecipeService recipeService;
+    @Autowired
+    public FoodPreferencesService(FoodPreferencesRepository foodPreferencesRepository) {
+        this.foodPreferencesRepository = foodPreferencesRepository;
```
wstrzykujemy przez konstruktor

```suggestion
    private final FoodPreferencesRepository foodPreferencesRepository;
   
    private final RecipeService recipeService;
    
    public FoodPreferencesService(FoodPreferencesRepository foodPreferencesRepository, RecipeService recipeService) {
        this.foodPreferencesRepository = foodPreferencesRepository;
        this.recipeService = recipeService;
```

szablewski reviewed (commented):
- pliki .json mozna juz usunac skoro macie juz podlaczona baze danych.
- brak testów.
-  setery tworzymy dopero wtedy gdy jestesmy pewni, że są potrzebne.
- wszystkie klasy które są rozszerzane przez Product mają określoną tę samą tabele "products" przez co wszystkie te rekordy właśnie tam wpadają, czy aby napewno tak ma właśnie być ? 
- metody zapisujące i odczytujące pliki json są zbędne.
-

agatamieniuk committed:
Delete json file and writeJson/getJson method.

agatamieniuk committed:
Delete App.class

agatamieniuk committed:
Delete implementation of CommandLineRunner class, exception and run(); method.

PDELIMATA committed:
some issues fixed.

PDELIMATA committed:
Merge remote-tracking branch 'origin/JJDZR6JTH-00' into JJDZR6JTH-00

agatamieniuk committed:
Change foodpreferences/set endpoint.

agatamieniuk committed:
Change request mapping in FridgeController

agatamieniuk committed:
Add final to fridgeService; and fridgeRepository; in FridgeController.class

agatamieniuk committed:
Add final to foodPreferencesService; in FoodPreferencesController.class

PDELIMATA committed:
fied issue with editing empty list of products in recipe.

PDELIMATA committed:
Merge remote-tracking branch 'origin/JJDZR6JTH-00' into JJDZR6JTH-00

agatamieniuk committed:
Change /new endpoint in Fridge.class to fridge/product

PDELIMATA committed:
fixed issues with FoodPreferencesController.

PDELIMATA committed:
Merge remote-tracking branch 'origin/JJDZR6JTH-00' into JJDZR6JTH-00

# Conflicts:
#	src/main/java/com/infoshareacademy/controller/FoodPreferencesController.java

agatamieniuk committed:
Delete taste enum.

agatamieniuk committed:
Change /search endpoint to /filtered-prodcts in RecipeController

agatamieniuk committed:
Change /new endpoint to /recipe in RecipeController

PDELIMATA committed:
Products tables names fixed.

PDELIMATA committed:
Merge remote-tracking branch 'origin/JJDZR6JTH-00' into JJDZR6JTH-00
